api/v1beta1/foundationdbcluster_types.go

```diff
@@ -2107,13 +2108,34 @@ const (
 	RedundancyModeUnset RedundancyMode = ""
 )
 
+// StorageEngine defines the storage engine for the database
+// +kubebuilder:validation:MaxLength=100
+type StorageEngine string
+
+const (
+	// StorageEngineSSD defines the storage engine ssd.
+	StorageEngineSSD StorageEngine = "ssd"
+	// StorageEngineSSD2 defines the storage engine ssd-2.
+	StorageEngineSSD2 StorageEngine = "ssd-2"
+	// StorageEngineMemory defines the storage engine memory.
+	StorageEngineMemory StorageEngine = "memory"
+	// StorageEngineMemory2 defines the storage engine memory-2.
+	StorageEngineMemory2 StorageEngine = "memory-2"
+)
+
 // DatabaseConfiguration represents the configuration of the database
 type DatabaseConfiguration struct {
 	// RedundancyMode defines the core replication factor for the database.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=single;double;triple
+	// +kubebuilder:default:double
 	RedundancyMode RedundancyMode `json:"redundancy_mode,omitempty"`
 
 	// StorageEngine defines the storage engine the database uses.
-	StorageEngine string `json:"storage_engine,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=ssd;ssd-1;ssd-2;memory;memory-1;memory-2;ssd-redwood-1-experimental;ssd-rocksdb-experimental;memory-radixtree-beta;custom
```

Comment:
I took the fields from: https://github.com/apple/foundationdb/blob/master/fdbclient/DatabaseConfiguration.cpp#L290-L312